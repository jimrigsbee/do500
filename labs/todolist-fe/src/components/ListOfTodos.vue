<template>
  <div>
    <md-list>
      <!-- TODO - change to an actual KEY when i connect to the DB -->
      <div v-for="item in todos" :key="item.id" >
        <TodoItem
          :todoItem=item
        ></TodoItem>
      </div>
    </md-list>
  </div>
</template>


<script>
import TodoItem from "@/components/TodoItem.vue";
import { mapGetters } from "vuex";

export default {
  name: "ListOfTodos",
  props: {},
  components: {
    TodoItem
  },
  computed: {
    ...mapGetters(["todos"])
  },
  created() {
    this.$store.dispatch("loadTodos");
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
</style>
