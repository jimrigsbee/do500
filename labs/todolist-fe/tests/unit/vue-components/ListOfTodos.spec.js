/* eslint-disable */
import { shallow, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import ListOfTodos from "@/components/ListOfTodos.vue";

import * as all from "../setup.js";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("ListOfTodos.vue", () => {
  let store;
  const todos = [
    {
      title: "Learn awesome things about Labs",
      completed: false,
      important: false
    }
  ];
  const actions = {
    loadTodos: jest.fn()
  };
  const getters = {
    todos: jest.fn()
  };
  beforeEach(() => {
    store = new Vuex.Store({
      state: {},
      propsData: { todos },
      actions,
      getters
    });
  });

  it("calls the loadTodos function from actionsjs when created", () => {
    const wrapper = shallow(ListOfTodos, { store, localVue });
    expect(wrapper).toBeTruthy();
    expect(actions.loadTodos).toHaveBeenCalled();
  });

  it("maps getters with todos when computed", () => {
    const wrapper = shallow(ListOfTodos, { store, localVue });
    expect(wrapper).toBeTruthy();
    expect(getters.todos).toHaveBeenCalled();
  });

  it("has the expected html structure", () => {
    const wrapper = shallow(ListOfTodos, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  });
});
