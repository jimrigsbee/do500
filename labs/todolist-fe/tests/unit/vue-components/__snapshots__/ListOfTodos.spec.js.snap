// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfTodos.vue has the expected html structure 1`] = `
<div>
  <ul
    class="md-list md-theme-default"
  />
</div>
`;
